File: configurator/orangepizero/shutdown-on-button-press/shutdown-on-button-press.js
var Gpio = require('onoff').Gpio;
var childProcess = require('child_process');

const gpioNumber = 10;

var button = new Gpio(gpioNumber, 'in', 'both');
var timer = null;

console.log('Monitoring GPIO' + gpioNumber);

function shutdown() {
  console.log('Shutting down!');
  childProcess.exec('sudo /bin/sh -c "poweroff"');
}

button.watch((err, value) => {
  if (timer) {
    clearTimeout(timer);
    timer = null;
  }

  if (!value) {
    timer = setTimeout(shutdown, 2000);
    console.log('Checking if button is held for more than 2 seconds ...');
  }
});

File: configurator/orangepizero/shutdown-on-button-press/shutdown-on-button-press.py
#!/bin/env python

import os
import sys
import time

gpioNumber = "6"
export = '/sys/class/gpio/export'
direction = '/sys/class/gpio/gpio' + gpioNumber + '/direction'
value = '/sys/class/gpio/gpio' + gpioNumber + '/value'
timeout = time.time()

def shutdown():
  print('Shutting down ...')
  os.system('/bin/sh -c "sudo poweroff"')
  sys.exit(0)

try:
  with open(export, 'w') as f:
    f.write(gpioNumber)
except IOError:
  pass

time.sleep(0.1)

with open(direction, 'w') as f:
  f.write('in')

with open(value, 'r') as f:
  print('Monitoring GPIO' + gpioNumber)
  while True:
    f.seek(0)
    newValue = int(f.read())

    if newValue:
      timeout = time.time()


    if (time.time() - timeout) > 2.0:
      shutdown()

    time.sleep(0.1)

